File: Javascript/tasks/caesarBoxCipherEncoding2.rb
def caesarBoxCipherEncoding2(message)
    (2...message.size).count { |n|
        encode(encode(message, n), n) == message }
end

def encode msg, n
    return nil if msg.nil? || msg.size % n != 0
    a = msg.chars.to_a
    z = []
    while a.size > 0
        z << a.shift(n)
    end
    z = z.transpose
    z.map(&:join).join
end

File: tasks/caesarBoxCipherEncoding2.py
def caesarBoxCipherEncoding2(message):
    r = 0
    for i in range(2, len(message)):
        if len(message) % i != 0:
            continue
        if encode(encode(message, i), i) == message:
            r +=1

    return r

def encode(msg, n):
    r = ''
    for i in range(n):
        for c in range(i, len(msg), n):
            r += msg[c]
    return r
